src/main/java/com/rsmart/kuali/kfs/cr/batch/service/impl/StaleCheckExtractServiceImpl.java

```diff
@@ -0,0 +1,330 @@
+            } else {
+                for (String inputFileName : inputFileNames) {
+
+                    if (StringUtils.isBlank(inputFileName)) {
+                        criticalError("getListOfFilesToProcess: One of the file names returned as ready to process [" + inputFileName
+                                + "] was blank.  This should not happen, so throwing an error to investigate.");
+                    }
+
+                    inputFileTypeMap.put(inputFileName, batchInputFileType);
+                }
+            }
+        }
+
+        return inputFileTypeMap;
+    }
+
+    protected List<String> loadStaleCheckBatchFile(String inputFileName, BatchInputFileType batchInputFileType) {
+        List<String> failedRowsErrors = new ArrayList<>();
+        byte[] fileByteContent = safelyLoadFileBytes(inputFileName);
+        LOG.info("loadStaleCheckBatchFile: Attempting to parse the file.");
+        
+        Object parsedObject = null;
+        try {
+            parsedObject = batchInputFileService.parse(batchInputFileType, fileByteContent);
+        } catch (ParseException e) {
+            String errorMessage = "loadStaleCheckBatchFile: Error parsing batch file: " + e.getMessage();
+            LOG.error(errorMessage, e);
+            throw new RuntimeException(errorMessage);
+        }
+        
+        if (!(parsedObject instanceof List)) {
+            String errorMessage = "loadStaleCheckBatchFile: Parsed file was not of the expected type.  Expected [" + List.class + "] but got [" + parsedObject.getClass() + "].";
+            criticalError(errorMessage);
+        }
+
+        List<StaleCheckBatchRow> staleChecks = (List<StaleCheckBatchRow>) parsedObject;
+        for (StaleCheckBatchRow staleCheck : staleChecks) {
+            String error = processStaleCheckBatchRow(staleCheck);
+            if (StringUtils.isNotBlank(error)) {
+                failedRowsErrors.add(error);
+            }
+        }
+        
+        return failedRowsErrors;
+    }
+
+    protected String processStaleCheckBatchRow(StaleCheckBatchRow staleCheckRow) {
+        LOG.info("processStaleCheckBatchRow: Starting processStaleCheckBatchRow for: " + staleCheckRow.getLogData());
+        CheckReconciliation checkReconciliation = checkReconciliationDao.findByCheckNumber(staleCheckRow.getCheckNumber());
+
+        String processingError = validateStaleCheckBatchRow(staleCheckRow, checkReconciliation);
+        if (StringUtils.isNotBlank(processingError)) {
+            return processingError;
+        }
+        checkReconciliation.setStatus(CRConstants.STALE);
+        checkReconciliation.setStatusChangeDate(dateTimeService.getCurrentSqlDate());
+        businessObjectService.save(checkReconciliation);
+
+        return processingError;
+    }
+
+    protected String validateStaleCheckBatchRow(StaleCheckBatchRow staleCheckRow, CheckReconciliation checkReconciliation) {
+        String failureMessage = KFSConstants.EMPTY_STRING;

```

Comment:
If you change failureMessage to be an ArrayList<String> here, and add elements to the list below, you can avoid all these string concatenations and keeping track of the index.  
At the end of validateStaleCheckBatchRow you can call a function to format all the elements in the array list to a single message, much like you are doing in appendFailureMessage